nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -12,76 +12,207 @@
 
-  // Get the filename to use from the parameter
-  get_parameter_or<std::string>("bt_xml_filename", bt_xml_filename_,
+  // Create the path that will be returned from ComputePath and sent to FollowPath
+  goal_ = std::make_shared<geometry_msgs::msg::PoseStamped>();
+  path_ = std::make_shared<nav2_msgs::msg::Path>();
+
+  // Create the blackboard that will be shared by all of the nodes in the tree
+  blackboard_ = BT::Blackboard::create<BT::BlackboardLocal>();
+
+  // Put items on the blackboard
+  blackboard_->set<geometry_msgs::msg::PoseStamped::SharedPtr>("goal", goal_);  // NOLINT
+  blackboard_->set<nav2_msgs::msg::Path::SharedPtr>("path", path_);  // NOLINT
+  blackboard_->set<nav2_lifecycle::LifecycleNode::SharedPtr>("node", node);  // NOLINT
+  blackboard_->set<std::chrono::milliseconds>("node_loop_timeout", std::chrono::milliseconds(10));  // NOLINT
+  blackboard_->set<bool>("initial_pose_received", false);  // NOLINT
+
+  // Get the BT filename to use from the node parameter
+  std::string bt_xml_filename;
+  get_parameter_or<std::string>(std::string("bt_xml_filename"), bt_xml_filename,
     std::string("bt_navigator.xml"));
 
-  // Read the input BT XML file into a string
-  std::ifstream xml_file(bt_xml_filename_);
+  // Read the input BT XML from the specified file into a string
+  std::ifstream xml_file(bt_xml_filename);
 
   if (!xml_file.good()) {
-    RCLCPP_ERROR(get_logger(), "Couldn't open input XML file: %s", bt_xml_filename_.c_str());
-    return TaskStatus::FAILED;
+    RCLCPP_ERROR(get_logger(), "Couldn't open input XML file: %s", bt_xml_filename.c_str());
+    return nav2_lifecycle::CallbackReturn::FAILURE;
   }
 
-  std::string xml_string((std::istreambuf_iterator<char>(xml_file)),
-    std::istreambuf_iterator<char>());
+  xml_string_ = std::string(std::istreambuf_iterator<char>(xml_file),
+      std::istreambuf_iterator<char>());
+
+  RCLCPP_DEBUG(get_logger(), "Behavior Tree file: '%s'", bt_xml_filename.c_str());
+  RCLCPP_DEBUG(get_logger(), "Behavior Tree XML: %s", xml_string_.c_str());
+
+  // Create the Behavior Tree from the XML input (after registering our own node types)
+  BT::Tree temp_tree = bt_->buildTreeFromText(xml_string_, blackboard_);
 
-  RCLCPP_INFO(get_logger(), "Behavior Tree file: '%s'", bt_xml_filename_.c_str());
-  RCLCPP_INFO(get_logger(), "Behavior Tree XML: %s", xml_string.c_str());
+  // Unfortunately, the BT library provides the tree as a struct instead of a pointer. So, we will
+  // createa new BT::Tree ourselves and move the data over
+  tree_ = std::make_unique<BT::Tree>();
+  tree_->root_node = temp_tree.root_node;
+  tree_->nodes = std::move(temp_tree.nodes);
+  temp_tree.root_node = nullptr;
 
-  // Create and run the behavior tree
-  NavigateToPoseBehaviorTree bt(shared_from_this());
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+BtNavigator::on_activate(const rclcpp_lifecycle::State &)
```

Comment:
should we add `/*state*/` to be consistent?  similar to `BtNavigator::on_configure(const rclcpp_lifecycle::State & /*state*/)`

Reply:
Fixed.